feat: navbar and responsive menu dialog redesign

conventoangelo opened the pull request:
Visual changes to the navigation bar of the header.

- Added hover styles.
- Made buttons feel more like buttons.
- Linked Get in Touch to UP CSI email. Renamed it to Contact Us.
- Removed redundant "Home" navigation. Moved functionality to CSI logo.
- Other minor changes.

I tested mobile responsiveness as much as I could. Made in participation of Hacktoberfest 🚀💻🫡.

ehrelevant commented on src/routes/NavBar.svelte:
```diff
@@ -10,62 +10,125 @@
     function closeMenu() {
         mobileMenu = false;
     }
+
```
```suggestion

```

I know this is a nitpick, but see prior comment. Formatter would normally flag this.

ehrelevant replied:
```suggestion

```

I know this is a nitpick, but see prior comment. Formatter would normally flag this.

ehrelevant commented on src/routes/NavBar.svelte:
```diff
@@ -10,62 +10,125 @@
-                <li class="m-4"><button on:click={closeMenu}>Get in Touch</button></li>
-            </ul>
-            <div class="flex w-full items-center justify-center">
+    <div
+        class="fixed right-0 top-0 block h-screen w-3/4 max-w-sm bg-blue-white lg:hidden dark:bg-black"
+    >
```
```suggestion
    <div
        class="fixed right-0 top-0 block h-screen w-full bg-blue-white sm:w-3/4 sm:max-w-sm lg:hidden dark:bg-black"
    >
```

I'm a bit unsure about the design of the mobile NavBar at the moment, At the very least, I think it should use the `w-full` for very small screens since it feels cramped otherwise.

VeeIsForVanana replied:
On this note, could we do something about the UP CSI label on the NavBar turning into two lines on smaller displays? Thanks!

conventoangelo replied:
I haven't taken the very small screens into account when making this. Noted on the UP CSI logo. Would add another commit regarding this later. 

Are you fully sold on the `w-full`?  I was going for the feeling that the user is just glossing over the navigation and still connected to the page. May I suggest doing `w-full md:w-3/4` or `sm:w-3/4` whatever works best for very small screens.

ehrelevant reviewed (changes requested):
Hello! Thank you and wonderful work on this PR, we really appreciate it! 

I just added some comments and slight nitpicks regarding running the formatter and linter, as well as some other minor things I noticed. As for whether or not this change to the NavBar design is appropriate and should be accepted will be up to @VeeIsForVanana to decide.

VeeIsForVanana commented on src/routes/NavBar.svelte:
```diff
@@ -1,5 +1,5 @@
 <script>
-    import { Moon, Sun } from '@steeze-ui/heroicons';
+    import { Moon, Sun, Bars3, XMark } from '@steeze-ui/heroicons';
```
It turned out I needed to approve the workflow run.

conventoangelo replied:
I tried to but I discarded the changes since a lot of other files (around 45 others) were changed. Also, if I remember correctly, sveltecheck in `pnpm fmt:fix` had an error.

VeeIsForVanana replied:
45 files changed from a `pnpm fmt:fix` run? That is interesting. @ehrelevant I'll note that @Ulyzses had a similar issue where `prettier` flagged and edited a whole bunch of files they had not modified. IIRC it was also ~40 files. 

When I ran the same command, I only had a single file modified. Someone please replicate. In the meantime, I've executed the `prettier` run myself and will be pushing the chore commit. Also, @conventoangelo could you send us a copy of a `pnpm fmt` run's logs here? 

@BastiDood you might find this interesting. Also requesting your review.

ehrelevant replied:
Okay I'll take note of that.

> When I ran the same command, I only had a single file modified. Someone please replicate.
I also ran the command on the branch earlier, and can confirm that only a single file was modified for me as well. Hmmm, wonder what's causing it.

BastiDood replied:
I got it. I forgot to add `.svelte-kit/` to the `.prettierignore`. I'll go submit a PR right now to fix that. ~~I need the Hacktoberfest points.~~ 😅

BastiDood replied:
This should be addressed by #84. For your approval @VeeIsForVanana. 🚀

VeeIsForVanana commented:
Requesting @BastiDood 's review since this is a notable departure from the original design.

BastiDood reviewed (changes requested):
> Requesting @BastiDood 's review since this is a notable departure from the original design.

I have a few reservations against the navigation bar specifically. Everything else is a very welcome improvement.

* I personally would discourage the blur effect on the navigation bar as its murkiness makes scrolling down the page quite distracting.
* I would also discourage the blur effect on the basis of accessibility. Blurred content behind text is a recipe for poor text contrast. As images fly behind the blurry navigation bar, there will be certain sections in the page where the navigation links will not be readable. This may be alleviated by choosing a different color palette, but generally speaking, we should avoid it altogether.

Despite this, however, I am not against the idea of a larger navigation bar. We can still take the happy medium between the original design and the proposed design by maintaining the floating navigation bar, but larger. See the [Excalidraw](https://excalidraw.com/) toolbar for an example of what I mean by a "larger [floating] navigation bar". I believe this compromise best preserves the spirit of Allaine's original design.

VeeIsForVanana commented:
Please merge in the latest `main`. Thanks!

conventoangelo commented:
Changed the style of the navbar to somewhat closer to Excalidraw and dropped the background blur effect. My bad, I wasn't aware that there was a design to be followed. Personally, I like the black navbar on white background as having white navbar on white background doesn't feel right. Same applies to the dark mode.

VeeIsForVanana commented:
Please refer to this [original mockup by Allaine](https://www.figma.com/file/gdObjB04j7BT36Q0lQ9GzL/UP-CSI-Website?type=design&node-id=203-2&mode=design)

VeeIsForVanana commented:
I'm personally not so sure about the stark white-on-black or vice versa. I honestly thought there was a bug with the dark/light modes when I first saw it. There might be a way to make this specific combination work, but right now for me it doesn't.

I do agree with the sentiment against black-on-black, but I think we could look for something with the colors that's closer in line with the original design. Otherwise, I do like the look of the hybrid between the original design and the proposed full-width navbar.

I'll return this PR to draft for the moment while we hash out the design. I do encourage the continuation of work on this PR, and would like to see it merged soon.

conventoangelo commented:
### UI Improvements:

* [`src/routes/NavBar.svelte`](diffhunk://#diff-f2af014e9d376284ca2aa4f3692aaef4d9dc8941cf6247eca4153c9004e15d91L2-R4): Refactored the navigation bar to include new icons (`Bars3`, `XMark`) and transitions (`fly`). Added a new mobile menu with improved styling and functionality. Updated the layout and styling of the navigation items and buttons.

### Layout Adjustments:

* [`src/routes/+layout.svelte`](diffhunk://#diff-cc69d6a97e4f62578028d872b8d5032f2786fe1eaa65735429286b375234168dL13-R16): Modified the header and main layout classes to improve alignment and spacing.

### Tailwind CSS Configuration:

* [`tailwind.config.ts`](diffhunk://#diff-655dc9e3d0aa561e3fa164bf48bd89cb0f5da65e0a567f8ebbf9dd791a0e7f40R38-R39): Added new color definitions for `light-gray` and `dark-gray` to the Tailwind CSS configuration.

ehrelevant commented on src/routes/NavBar.svelte:
```diff
@@ -12,60 +13,122 @@
-        </li>
-    </ul>
-</nav>
+<div
+    class="flex items-center justify-between rounded-full border-2 border-light-gray bg-csi-white/90 p-1 font-dm text-csi-black shadow-xl backdrop-blur-xl lg:mx-24 dark:border-dark-gray dark:bg-csi-black/90 dark:text-csi-white"
+>
```
```suggestion
<div
    class="flex items-center justify-between rounded-full border-2 border-light-gray bg-csi-white/90 p-1 font-dm text-csi-black shadow-lg backdrop-blur-xl lg:mx-24 dark:border-dark-gray dark:bg-csi-black/90 dark:text-csi-white"
>
```
This is very minor, I just think that `shadow-xl` would be too much here especially since it already has a border. `shadow-lg` should suffice.

ehrelevant commented on src/routes/NavBar.svelte:
```diff
@@ -12,60 +13,122 @@
-                <li class="m-4"><button on:click={closeMenu}>Get in Touch</button></li>
-            </ul>
-            <div class="flex w-full items-center justify-center">
+    <div
+        class="fixed inset-y-2 right-2 block h-[calc(100vh-1rem)] w-5/6 rounded-2xl bg-blue-white/95 font-dm text-xl backdrop-blur-2xl sm:w-3/4 sm:max-w-sm lg:hidden dark:bg-black/95 dark:text-csi-white"
+        transition:fly={{ x: 300, duration: 150 }}
+    >
```
```suggestion
    <div
        class="fixed inset-2 block rounded-2xl bg-blue-white/95 font-dm text-xl backdrop-blur-2xl sm:left-auto sm:w-3/4 sm:max-w-sm lg:hidden dark:bg-black/95 dark:text-csi-white"
        transition:fly={{ x: 300, duration: 150 }}
    >
```
I don't think `h-[calc(100vh-1rem)]` would do anything with both `fixed` and `inset-y-2` set here.

That aside, similar to what I've mentioned in a prior review, I think the mobile NavBar still looks awkward on small screens, even with `w-5/6`. These changes make it so that it instead extends to the other side of the screen for smaller screens while keeping the slight margin of `0.5rem`.

conventoangelo replied:
`sm:left-auto` was actually the property I was looking for. But I don't use `-auto` that much which is why I settled for w-5/6. Will surely use this instead. Thanks!

ehrelevant reviewed (changes requested):
Didn't expect I'd be asked for a review, but alright then. Thanks for the updated PR, your updated NavBar looks great!

Just writing down a few minor nitpicks again. While I won't comment on the translucent/blurred NavBar, just a heads up that the other reviewers might. That said, I think that the opacity and blur seem high enough for the problems @BastiDood previously stated to not be that big of an issue.

VeeIsForVanana commented:
I have to concur with @ehrelevant , this new version looks amazing! 🚀

I'm all clear with the current approach and I'm pretty good on how it looks on mobile. The solution to my previous issues with the dark mode NavBar blending too much into the background is beautifully subtle yet effective. My approval is pending on @BastiDood 's remaining comments.

Thanks again for this PR, the improvement looks great!

BastiDood reviewed (approved):
This is beautiful! I love it. Thanks @conventoangelo! Really **great work** on applying the feedback and the spirit of the design. 🚀

@VeeIsForVanana This man is ready for the service projects. Official seal of approval from an alumni exec (ex-exec???).

VeeIsForVanana reviewed (approved):
All clear. LGTM! 🚀🚀🚀

Thanks again @conventoangelo !
